x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/whitelist/InternalEqlScriptUtils.java

```diff
@@ -33,6 +36,30 @@
     InternalEqlScriptUtils() {
     }
 
+    public static <T> Boolean multiValueDocValues(Map<String, ScriptDocValues<T>> doc, String fieldName, Predicate<T> script) {
+        if (doc.containsKey(fieldName)) {
+            ScriptDocValues<T> docValues = doc.get(fieldName);
+            if (docValues.isEmpty() == false) {
+                for (int i = 0; i < docValues.size(); i++) {
+                    T value = docValues.get(i);
+                    if (script.test(value)) {
+                        return true;
+                    }
+                }
```

Comment:
This can be simplified by letting the else fall through to avoid repeating the script.test:

```java
if (doc.containsKey()...
  if (docValues.is)
   ...
  }
}
return script.test(null);
```

Reply:
Thanks. I've simplified it even further.